### ISSUE TITLE
Order is not confirmed after making the payment

### ISSUE THREAD

**Issue by nggopakumar** [willxinc/odoo-payment_moneris#1]
After making the payment order is not confirmed and the item is not removed from the cart in ODOO v8.
![screenshot from 2016-05-04 13 11 03](https://cloud.githubusercontent.com/assets/7919826/15007875/bf83ba3c-11f9-11e6-815c-638ca9c5f50d.png)
![screenshot from 2016-05-04 13 11 21](https://cloud.githubusercontent.com/assets/7919826/15007876/c1517dea-11f9-11e6-8915-4b066e9a162b.png)

**Comment by willxinc** [comment]
From what I can see since the order isn't confirming, the cart isn't being cleared. Even though the website says that the order is confirmed, on the server itself there's likely an issue. 
Are there any logs you could provide?
What does the payment configuration look like?
Also do you know what build of odoo you're running?, and in what system?

**Comment by nggopakumar** [comment]
I have set up a developer account in moneris and configured the ps_store_id and hpp_key in ODOO Version 8.0-20160205 in payment acquirers. I am running ODOO in ubuntu 14.04 LTS. The redirection URL from moneris after payment is something like http://test.testodoxxxneris.com/shop/confirmation

**Comment by willxinc** [comment]
Oh, the redirection URL is wrong, I probably should have mentioned somewhere, what that needs to be.
Proper config should be:
Approval URL: https://SITEURLHERE/payment/moneris/dpn
Declined URL: https://SITEURLHERE/payment/moneris/cancel

Use "Enhanced Cancel" enabled

Under Configure Response Fields, ECI and txn_number enabled.

Under security features, Transaction Verification enabled, and set to be displayed as key/value pairs on moneris's server.

**Comment by willxinc** [comment]
Hey there, any update on this?

**Comment by nggopakumar** [comment]
Thank you for the reply. It worked. The problem was with the Approval and Declined URL configuration. By the way, is this module available for ODOO 9?

**Comment by willxinc** [comment]
I haven't looked into what's changed in 9 much, but it should just work, unless the changed how their e-commerce model behaves. Were there any particular errors when trying it with 9?

**Comment by nggopakumar** [comment]
Yes, there are some changes in the payment module of ODOO 9's 'render' method which causes errors.
